metadata-io/src/main/java/com/linkedin/metadata/entity/DeleteEntityService.java

```diff
@@ -247,11 +248,10 @@ private void deleteAspect(Urn urn, String aspectName, RecordTemplate prevAspect)
         proposal.setEntityType(urn.getEntityType());
         proposal.setAspectName(aspectName);
 
-        final AuditStamp auditStamp = new AuditStamp().setActor(UrnUtils.getUrn(Constants.SYSTEM_ACTOR)).setTime(System.currentTimeMillis());
-        final EntityService.IngestProposalResult ingestProposalResult = _entityService.ingestProposal(proposal, auditStamp, false);
-
-        if (!ingestProposalResult.isDidUpdate()) {
-            log.error("Failed to ingest aspect with references removed. Before {}, after: null, please check MCP processor"
+        RollbackResult rollbackResult = _entityService.deleteAspect(urn.toString(), aspectName, new HashMap<>(), true);
+        assert rollbackResult != null;
```

Comment:
Assert statement should get wrapped into below conditional and proposal is no longer used except for the error message which is invalid as it's not being used.